### ISSUE TITLE
Feature request: Support Qubicle project files .qbcl

### ISSUE THREAD

**Issue by tostc** [guillaumechereau/goxel#271]
Hey,

this is less of a feature request and more of a note. I reverse engineered Qubicle's .qbcl format. The documentation can be found [here](https://gist.github.com/tostc/7f049207a2e5a7ccb714499702b5e2fd).

**Comment by mgerhardy** [comment]
Based on that information I did an implementation for vengi already - might serve as inspiration, too ;) https://github.com/mgerhardy/vengi/blob/master/src/modules/voxelformat/QBCLFormat.cpp

**Comment by guillaumechereau** [comment]
Yes would be nice to have.  Do you have some example files I could try?

**Comment by tostc** [comment]
Hey,

I used the demo files from the Qubicle demo on Steam. https://store.steampowered.com/app/454550/Qubicle_Voxel_Editor/

**Comment by guillaumechereau** [comment]
Thanks for the info.  By the way, I didn't know about Vengi but it looks very impressive!  I haven't tried it yet, but from the screenshots it looks much more advanced than goxel.

**Comment by guillaumechereau** [comment]
I just pushed a file that adds basic support for the format.  Thanks for the work doing the reverse engineering!

**Comment by tostc** [comment]
No problem. Reverse engineering the file format was a lot of fun.

**Comment by PhilT** [comment]
Hey there, I'm trying to use this feature but when I browse to the directory of qbcl files I don't see them to be able to import. I'm on Linux. Any ideas? Thanks.

**Comment by guillaumechereau** [comment]
Hello @PhilT, are you building the app yourself?

**Comment by tostc** [comment]
At line 193 in the qubicle2.c file is a little mistake. The variable r is at this point 4 and not 1
https://github.com/guillaumechereau/goxel/blob/ec31d198501e2e0b13294177fffc2b3f2bc16ee5/src/formats/qubicle2.c#L193

since on the line above count and elementsize are in the opposite order, so that fread returns 4 (count) and not 1 (elementsize)

https://github.com/guillaumechereau/goxel/blob/ec31d198501e2e0b13294177fffc2b3f2bc16ee5/src/formats/qubicle2.c#L192

I should also mention, that the signature ist not "QBCL\<SPACE\>", but "QBCL". This shouldn't be a problem here, since only the first four bytes are compared, but for the completeness I should mention it.

**Comment by mgerhardy** [comment]
> I've tried loading an old `.qbcl` file on Windows but got an error on Invalid magic (line 193) as pointed out in [#271 (comment)](https://github.com/guillaumechereau/goxe
> Would be nice to have stable support for this format, I have a collection of old stuff that I'd like to continue to work with on Goxel.

You could use vengi-voxconvert to convert them to the gox format to continue using goxel

See e.g. http://vengi-voxel.github.io/vengi/voxconvert/Examples/#convert-to-multiple-different-formats
